pymc/smc/smc.py

```diff
@@ -311,27 +301,16 @@ def _posterior_to_trace(self, chain=0) -> NDArray:
 
         This method should not be overwritten.
         """
-        lenght_pos = len(self.tempered_posterior)
-        varnames = [v.name for v in self.variables]
+        return self.particles.to_trace(chain)
 
-        with self.model:
-            strace = NDArray(name=self.model.name)
-            strace.setup(lenght_pos, chain)
-        for i in range(lenght_pos):
-            value = []
-            size = 0
-            for varname in varnames:
-                shape, new_size = self.var_info[varname]
-                var_samples = self.tempered_posterior[i][size : size + new_size]
-                # Round discrete variable samples. The rounded values were the ones
-                # actually used in the logp evaluations (see logp_forw)
-                var = self.model[varname]
-                if var.dtype in discrete_types:
-                    var_samples = np.round(var_samples).astype(var.dtype)
-                value.append(var_samples.reshape(shape))
-                size += new_size
-            strace.record(point={k: v for k, v in zip(varnames, value)})
-        return strace
+    def _tempered_posterior_p(self, samples):
```

Comment:
I am not sure this function is a good abstraction. It's not clear from the name that it will return 3 objects (perhaps add type hints?)

Also the internal names don't make sense out of their old context (e.g., samples -> prop). Probably better `for sample in samples`

Since it is its own function, it seems fine to give lp and pp longer names.